crypto/rsa/rsa_ossl.c

```diff
@@ -476,13 +483,91 @@ static int rsa_ossl_private_decrypt(int flen, const unsigned char *from,
         if (!rsa_blinding_invert(blinding, ret, unblind, ctx))
             goto err;
 
+    /*
+     * derive the Key Derivation Key from private exponent and public
+     * ciphertext
+     */
+    if (!(rsa->flags & RSA_FLAG_EXT_PKEY)) {
+        BIGNUM *d = BN_new();
```

Comment:
I still suggest making it a function-level variable

Reply:
And even more, a BN_CTX_get() will let you not worry about freeing it

Reply:
done

Reply:
yeah, that doesn't work, because it shadows the `rsa->d` value, we need to free it before `rsa->d` is used, otherwise we get memory leaks

Reply:
AFAIK, no. If BIGNUM is created via BN_CTX_new, you shouldn't manage memory explicitly.

Reply:
But unfortunately the BN_with_flags() semantics is like this. It cannot be used with bignum from ctx as the destination.